packages/@aws-cdk/aws-lambda/test/integ.lambda.filesystem.ts

```diff
@@ -0,0 +1,74 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as efs from '@aws-cdk/aws-efs';
+import * as cdk from '@aws-cdk/core';
+import * as lambda from '../lib';
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'aws-cdk-lambda-1');
+
+const vpc = new ec2.Vpc(stack, 'Vpc', {
+  maxAzs: 3,
+  natGateways: 1,
+});
+
+const fileSystem = new efs.FileSystem(stack, 'Efs', {
+  vpc,
+  throughputMode: efs.ThroughputMode.PROVISIONED,
+  provisionedThroughputPerSecond: cdk.Size.mebibytes(1024),
+});
+
+fileSystem.connections.allowDefaultPortInternally();
```

Comment:
OK the integ test confirms my previous comment about the security group. This is needed and not documented. Are you sure there's no other solution to offer a "it just works" experience for the user?

Reply:
Let's find one. This is not good enough.

Reply:
We need allow Lambda function connect to the EFS on TCP 2049 and we probably have some options:

Option 1:   

```ts
accessPoint.filesystem.connections.allowDefaultPortFrom(this);
```
Result: `cdk synth` is good, but on `cdk deploy` we got circular dependency error

>Circular dependency between resources: [MyLambdaServiceRole4539ECB6, MyLambdaCCE802FB, MyLambdaSecurityGroup1E71A818, EfsEfsSecurityGroupfromawscdklambda1MyLambdaSecurityGroup86B085EE20490D9864A8]

Option 2:

We do the `accessPoint.filesystem.connections.allowDefaultPortInternally();` in `lambda.Function` construct for  users.  This works. But are we supposed to update the efs security group in the Lambda construct for users?

And any other possible solutions?

Reply:
The problem: you're adding a dependency on the whole `FileSystem` construct. This means that everything in the `Function` construct gets a `DependsOn` on **everything** in the `FileSystem` construct. So the security group created inside the `Function` construct gets a `DependsOn` on the security group created inside the `FileSystem` construct. But the security of the file system references the security group of the Lambda (for the ingress rule). There's no real circular dependency, you are creating it by adding a dependency on the whole `FileSystem`, which is not needed.

The solution: from the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-filesystemconfigs) we see that the only real dependency the function requires is a dependency on the mount targets (`AWS::EFS::MountTarget`). I was able to successfully deploy an integ test using the `allowToDefaultPort()` solution by exposing the mount targets in the `FileSystem` construct and using this as the `dependency` in the `fromEfsAccessPoint()`. In order to quickly test it I exposed an array of `CfnMountTarget`. The clean solution would be to implement a `mountTargetsAvailable: IDependable` in `FileSystem`.

Reply:
@jogold Thanks for the valuable insight! I am so lucky you folks staying with me in this PR. Let me try it.

Reply:
@jogold 

Now the challenge is `accessPoint.filesystem` is actually `IFileSystem`. Seems it doesn't help if we expose the `CfnMountTarget` on `FileSystem` as we still can't access them from the `accessPoint`.  What do you think?

Reply:
You can use `instanceof` to check it it’s a `FileSystem` and then reference properties that are only available there. 

Please expose the underlying L1 as an IDependable not as the L1 type.

Reply:
This is comparable to the `internetConnectivityEstablished` in `aws-ec2`:
https://github.com/aws/aws-cdk/blob/916706b4c20a222da282dd62f46ceb384adb3f35/packages/%40aws-cdk/aws-ec2/lib/vpc.ts#L97-L100